daq/faucet_state_collector.py

```diff
@@ -165,9 +165,42 @@ def get_switch_raw(self, switch_name):
     def get_stack_topo(self):
         """Returns formatted topology object"""
         topo_map = {}
-        topo_obj = self.topo_state
         with self.lock:
-            for link in topo_obj.get(TOPOLOGY_GRAPH, {}).get("links", []):
+            config_obj = self.system_states.get(FAUCET_CONFIG, {}).get(DPS_CFG, {})
+            links = self.topo_state.get(TOPOLOGY_GRAPH, {}).get("links", [])
+            path_to_root = self.topo_state.get(ROOT_PATH, {})
+            for dp, dp_obj in config_obj.items():
+                for iface, iface_obj in dp_obj.get("interfaces", {}).items():
+                    dp_s = iface_obj.get("stack", {}).get("dp")
+                    port_s = str(iface_obj.get("stack", {}).get("port"))
+                    if dp_s and port_s:
+                        link_obj = {}
+                        if dp+":"+iface < dp_s+":"+port_s:
+                            link_obj["switch_a"] = dp
+                            link_obj["port_a"] = iface
+                            link_obj["switch_b"] = dp_s
+                            link_obj["port_b"] = port_s
+                            key = dp+":"+iface+"-"+dp_s+":"+port_s
+                        else:
+                            link_obj["switch_b"] = dp
+                            link_obj["port_b"] = iface
+                            link_obj["switch_a"] = dp_s
+                            link_obj["port_a"] = port_s
+                            key = dp_s+":"+port_s+"-"+dp+":"+iface
+                        topo_map[key] = link_obj
+                        link_obj["status"] = "DOWN"
```

Comment:
better to put this in an else clause -- lean towards only setting a variable once -- rather than setting it first and then changing it to something else.  Same with the bit below ("UP") -- right now, it's kinda hard to trace through and exactly know what the resulting value is.  You might want to create a helper function for 'is_link_up' that loops through the links and returns true/false, which would make the logic overall more straightforward...